Redesign enterprise tile

jingcheng16 opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->
Changes:
- Remove the descriptors below the stat, make that description a tooltip to avoid redundancy. ( Not every tile have a tooltip. Only the one that might cause confusion will have. )
- Change the button style to [default style](https://staging.commcarehq.org/styleguide/b5/molecules/buttons/#most-buttons)
- Change the stat to the non-bold thickness, the text size is big enough to negate the need for making it bold

Before:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/42130bcf-0217-4a4b-aea2-860065c4bff7" />

After:
![image](https://github.com/user-attachments/assets/b0d07e67-6dba-4d4e-bb3d-5d7b8ed370b8)

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
Ticket: https://dimagi.atlassian.net/browse/SAAS-16428

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Mainly appearance change.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->



### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

jingcheng16 committed:
Have a help icon next to the metric number

jingcheng16 committed:
Prettify

jingcheng16 committed:
Use default button style

per Sud's suggestion

jingcheng16 committed:
Change metric to the non-bold thickness

Sud: the text size is big enough to negate the need for making it bold

jingcheng16 committed:
Remove total_description for some straightforward metrics

jingcheng16 committed:
Rename API Usage tile and remove total_description

orangejenny commented on corehq/apps/enterprise/templates/enterprise/partials/project_tile.html:
```diff
@@ -8,10 +8,20 @@
+        {% if report.total_description %}
+          <div class="position-absolute start-100" style="top: 25%">
+            <span class="hq-help-template" data-content="{{ report.total_description }}" data-placement="right">
+              <i class="fa fa-info-circle"></i>
```
What would you think of putting the icon next to the header text instead of the number? It's more secondary/tertiary information than primary information.

jingcheng16 replied:
I feel putting the icon next to the header text implies this description is to explain the header text, not the stat there anymore. But maybe I can move the icon to next to the header text, and then in the tooltip, I included a description of the stat, also a description of the report they will get. I'll bring it up with saas-product. Thank you!

orangejenny commented on corehq/apps/enterprise/templates/enterprise/partials/project_tile.html:
```diff
@@ -8,10 +8,20 @@
+          <i class="fa fa-spin fa-spinner"></i>
+        </div>
+        {% if report.total_description %}
+          <div class="position-absolute start-100" style="top: 25%">
```
The `top: 25%` style could be a `top-25` class, right?

jingcheng16 replied:
Bootstrap only have `top-0`, `top-50` and `top-100`, so I have to use style: https://getbootstrap.com/docs/5.0/utilities/position/#arrange-elements

orangejenny replied:
aha, got it

biyeun replied:
can override the sass map to include `top-25`
https://getbootstrap.com/docs/5.3/utilities/position/#sass-maps
also i would prefer `end-0` instead of `start-100`

jingcheng16 replied:
override in 9119c9f35d750088df0342db393871a1ec803986
`start-100` put the left edge of the element at the right edge of the parent element
`end-0` put the right edge of the element at the left edge of the parent element
So they're not exchangeable...

jingcheng16 commented on corehq/apps/enterprise/templates/enterprise/partials/project_tile.html:
```diff
@@ -1,34 +1,53 @@
+              data-content="{{ report.total_description }}"
+              data-placement="right"
+            >
+              <i class="fa fa-info-circle"></i>
```
A note to myself: The span is already use the `hq-help-template`, the `fa-info-circle` is not required.

jingcheng16 committed:
Override bootstrap position values

Fix a small thing as well: <i> element will be override by "hq-help-template" anyway so remove it

jingcheng16 committed:
Display help icon until the stat is loaded

jingcheng16 commented:
@biyeun @mjriley This PR is ready for review again.

jingcheng16 committed:
"Bootstrap 5 Migration - Rebuilt diffs"

jingcheng16 committed:
Merge branch 'master' into jc/redesign-enterprise-tile

jingcheng16 committed:
Remove descriptor for DataExportReport
